docs/migration.md

```diff
@@ -0,0 +1,546 @@
+
+        switch (messageText) {
+            case "/ping":
+                this.messageService.send(streamId, "pong");
+                break;
+            case "/pong":
+                this.messageService.send(streamId, "ping");
+                break;
+            default:
+                this.messageService.send(streamId, "Sorry, I don't understand!");
+                break;
+        }
+    }
+}
+
+
+@Component
+public class RealTimeEventComponent {
+
+    private final PingPongBotService pingPongBotService;
+
+    public RealTimeEventComponent(PingPongBotService pingPongBotService) {
+        this.pingPongBotService = pingPongBotService;
+    }
+
+    @EventListener
+    public void onMessageSent(RealTimeEvent<V4MessageSent> event) {
+        this.pingPongBotService.handleIncomingMessage(event.getSource().getMessage,
+                StreamType.TypeEnum.formValue(event.getSource().getMessage.getStream.getStreamType()));
+    }
+}
+```
+</td>
+</tr>
+</table>
+
+An example of non framework based project using `SymphonyBdk` to retrieve BDK services:
+````java
+@Slf4j
+public class GreetingsAllRoomsBot {
+
+  public static void main(String[] args) throws Exception {
+
+    final SymphonyBdk bdk = new SymphonyBdk(loadFromSymphonyDir("config.yaml"));
+
+    // list all rooms
+    Stream<StreamAttributes> rooms = bdk.streams().listAllStreams(new StreamFilter());
+    
+    rooms.forEach(streamAttributes -> {
+          // send message to room
+          bdk.messages().send(streamAttributes.getId(), "Hello world!");
+          log.info("Message sent to room with: id:{}, name:{}", streamAttributes.getId(),
+              streamAttributes.getRoomAttributes().getName());
+    });
+    
+    bdk.datafeed().start();
+  }
+}
+````
+> A list of BDK available services can be found [here](./fluent-api.md)
+
+## Event listeners
+Java BDK 2.0 comes with a simplified way to handle event listeners.
+<table>
+<tr>
+<th>Java BDK 1.0</th>
+<th>Java BDK 2.0</th>
+</tr>
+<tr>
+<td>
+
+In Java BDK 1.0, the bot had to implement 3 listeners classes: 
+- one for IM (1 to 1 conversation)
+- one for MIM (room)
+- one for Symphony elements
+```java
+@Slf4j
+@Service
+public class ElementsListenerImpl implements ElementsListener {
+    public void onElementsAction() {...}
+}
+
+@Slf4j
+@Service
+public class ElementsListenerImpl implements ElementsListener {
+    public void onIMMessage() {...}
+}
+
+@Slf4j
+@Service
+public class RoomListenerImpl implements RoomListener {
+    public void onRoomMessage() {...}
+}
+```
+</td>
+<td>
+
+In Java BDK 2.0, only one component `RealTimeEventComponent` has to be implemented with two methods having `@EventListener` annotation: This works provided we have the correct parameters with correct types. The 3 classes can be factored in one single component. *(The example below uses a Spring Boot based project)* 
+```java
+public class RealTimeEventComponent {
+    @EventListener
+    public void onMessageSent() {...}
```

Comment:
Please add argument `RealTimeEvent<V4MessageSent> event` (because it must have the correct type in order to work)